my_enumerable.rb

```diff
@@ -0,0 +1,124 @@
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    idx = to_a
+    idx.size.times do |index|
+      yield idx[index]
+    end
+
+  end
```

Comment:
- [ ] When a block is given, and `my_each` is called on an array, it should return the array itself after calling the given block once for each element in the array. 

```ruby 
array = [1,2,3,4,56,6,7,53,23,45,1]
block = proc { |num| num < (1 + 9) / 2 }
array.each(&block ) === array.my_each(&block ) => false
```

- [ ] When a block is given, and `my_each` is called on a range, it should return the range itself after calling the given block once for each element in the range. 

```ruby 
range = Range.new(5,50)
block =proc { |num| num < (1 + 9) / 2 }
range.my_each(&block ) === range.each(&block )  => false
```